src/NServiceBus.Core/Routing/Publishing/StorageDrivenDispatcher.cs

```diff
@@ -0,0 +1,70 @@
+﻿namespace NServiceBus.Routing.Publishing
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using NServiceBus.ConsistencyGuarantees;
+    using NServiceBus.DeliveryConstraints;
+    using NServiceBus.Pipeline;
+    using NServiceBus.TransportDispatch;
+    using NServiceBus.Transports;
+    using NServiceBus.Unicast.Messages;
+    using NServiceBus.Unicast.Subscriptions;
+    using NServiceBus.Unicast.Subscriptions.MessageDrivenSubscriptions;
+
+    class StorageDrivenDispatcher : DispatchStrategy
+    {
+        public StorageDrivenDispatcher(ISubscriptionStorage subscriptionStorage, MessageMetadataRegistry messageMetadataRegistry)
+        {
+            this.subscriptionStorage = subscriptionStorage;
+            this.messageMetadataRegistry = messageMetadataRegistry;
+        }
+
+
+        public override void Dispatch(IDispatchMessages dispatcher, OutgoingMessage message, RoutingStrategy routingStrategy, ConsistencyGuarantee minimumConsistencyGuarantee, IEnumerable<DeliveryConstraint> constraints, BehaviorContext currentContext)
+        {
+            var currentConstraints = constraints.ToList();
+
+            var toAllSubscribersStrategy = routingStrategy as ToAllSubscribers;
```

Comment:
I'm game, would you like to take a stab at that? (since I'm not 100% sure exactly how you want it)